// actions/tiker.js
import axios from 'axios'


export const GET__TIKER__LOADING = 'GET__TIKER__LOADING'
export const GET__TIKER__SUCCESS = 'GET__TIKER__SUCCESS'
export const GET__TIKER__ERROR = 'GET__TIKER__ERROR'


//onGetTiker
function findCurrenciesAndCoins(tikers) {
  let currencies = [];
  let dataCurrencies = [];

  //find unique currencies
  for (let tiker in tikers) {
    currencies.push(tiker.split('_')[0]); // add all currencies from the tickers
  }
  currencies = [...new Set(currencies)]; // unique currencies
  //END find unique currencies


  //find all the coins for each currencie
  for (let i = 0; currencies.length > i; i++) {
    let coins = [];

    for (let tiker in tikers) {
      if (String(currencies[i]) === String(tiker.split('_')[0])) {
        coins.push(tiker.split('_')[1])
      }
    }

    dataCurrencies[i] = {
      name: currencies[i],
      coins
    }
  }
  //END find all the coins for each currencie

  return dataCurrencies;
}


export const onGetTiker = () => async dispatch => {
  dispatch({ type: GET__TIKER__LOADING })

  try {
    const tiker = await axios.get('https://poloniex.com/public?command=returnTicker').catch(error => { throw error })

    dispatch({
      type: GET__TIKER__SUCCESS,
      currencies: findCurrenciesAndCoins(tiker.data)
    })
  } catch (error) {
    dispatch({
      type: GET__TIKER__ERROR
    })
  }
}
//END onGetTiker
