// containers/TradeHistory/index.js
import React, { Component } from 'react'
import { connect } from 'react-redux'
// import PropTypes from 'prop-types'
// import FontAwesomeIcon from '@fortawesome/react-fontawesome'
// import faCog from '@fortawesome/fontawesome-free-solid/faCog'


import './index.css';
import { onGetTiker } from 'actions/tiker'
import { onGetTradeHistory } from 'actions/tradeHistory'
//Components
// import PagesList from 'components/Pages'


class TradeHistory extends Component {
  constructor() {
    super()

    this.state = {
      seletedCurrencie: ''
    }
  }

  componentDidMount() {
    this.props.onGetTiker();
  }

  onGetTradeHistory = () => {
    this.props.onGetTradeHistory()
  }

  handleSelectedCurrencie = (e) => {
    const currencie = e.target.value;

    this.setState({ seletedCurrencie: currencie })
  }

  render() {
    return (
      <div>
        <br />
        <div>
          <label>Currencie: </label>
          <select onChange={this.handleSelectedCurrencie}>
            <option value=''>Select currency</option>
            {
              this.props.currencies.map((item, num) => {
                return <option key={num} value={item.name}>{item.name}</option>
              })
            }
          </select>

          &emsp;

          <label>Coin: </label>
          <select>
            <option value=''>Select coin</option>
            {
              this.props.currencies.filter(item =>
                String(item.name) === String(this.state.seletedCurrencie)
              ).map(item => {
                return item.coins.map((coin, num) => {
                  return <option key={num} value={coin}>{coin}</option>
                })
              })
            }
          </select>
        </div>

        <br />
        ======================================
        <br /> <br />


        <button onClick={this.onGetTradeHistory}>Get trade history</button>
        {this.props.tradeHistory.length ?
          <div>
            <ul>
              {
                this.props.tradeHistory.map((item, num) => {
                  return (
                    <li key={num}>{item.date}</li>
                  )
                })
              }
            </ul>
            Всего записей: {this.props.tradeHistory.length}
          </div>
          :
          null
        }
      </div>
    )
  }
}


// TradeHistory.propTypes = {
//   data: PropTypes.string,
// }


const mapStateToProps = (state) => ({
  tradeHistory: state.tradeHistory.data,
  tiker: state.tiker,
  currencies: state.tiker.currencies,
})

const mapDispatchToProps = ({
  onGetTradeHistory,
  onGetTiker
})

export default connect(mapStateToProps, mapDispatchToProps)(TradeHistory);
